app/src/main/java/com/asfoundation/wallet/onboarding/use_cases/PendingPurchaseFlowUseCase.kt

```diff
@@ -21,27 +21,22 @@ class PendingPurchaseFlowUseCaseImpl @Inject constructor(
     var startModeResult: StartMode.PendingPurchaseFlow? = null
     val cachedTransaction = cachedTransaction.getCachedTransaction().blockingGet()
     val cachedBackupKey = cachedBackup.getCachedBackup().blockingGet()
-    if (cachedBackupKey != null) {
+    val integrationFlow = if (cachedBackupKey != null) {
```

Comment:
I think this would be cleaner:

```
val integrationFlow = when {
      (cachedBackupKey != null && cachedTransaction?.type?.uppercase() == PAYMENT_TYPE_SDK) || cachedTransaction?.callbackUrl == null ->
        "sdk"

      else -> "osp"
    }
```